app/controllers/pg_hero/home_controller.rb

```diff
@@ -275,6 +275,22 @@ def connections
 
       @connections_by_database = group_connections(@connection_sources, :database)
       @connections_by_user = group_connections(@connection_sources, :user)
+
+      @citus_enabled = @database.citus_enabled?
+      if @citus_enabled
+        @citus_worker_count = @database.citus_worker_count
+        @citus_worker_connection_sources = @database.citus_worker_connection_sources(@citus_worker_count)
+        @citus_worker_total_connections = @database.citus_worker_total_connections(@citus_worker_count)
+  
+        @citus_worker_connections_by_database = Array.new(@citus_worker_count)
+        @citus_worker_connections_by_user = Array.new(@citus_worker_count)
+        count_citus_workers = 0
+        while count_citus_workers < @citus_worker_count
```

Comment:
In ruby, just avoid these kinds of loops as much as possible. Instead, you can use each. It is one of the strongest features of ruby and simplifies the code a lot.